### PR TITLE
KB-3801, KB-3802, KB-3788, KB-3789, KB-3785

### PR DESCRIPTION
**Pull request by NandiniAV**
Edit form inside user card,
UI updates,
Approvals tab integration & listing

### CODE REVIEW

File: project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html [hunk 1]
```diff
@@ -48,187 +41,240 @@
...
-                            <ng-container>
-                              <ul class="roles-list margin-0">
-                                <li *ngFor="let role of orguserRoles" class="margin-bottom-s">{{role}}</li>
-                              </ul>
-                            </ng-container>
-                          </div>
-                        </div>
-                      </div>
-                    </mat-card-content>
+      <div class="panel-body profile-box" *ngIf="user.enableEdit">
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html]: please do optional check

File: project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html [hunk 2]
```diff
@@ -48,187 +41,240 @@
...
+            <p class="float-right mat-body-2"><mat-slide-toggle>Mark this user as non-verified</mat-slide-toggle></p>
+            <div class="flex flex-row margin-top-m">
+              <div class="md:w-1/3">
+                <label class="margin-remove-bottom form-label required">Designation</label>
+                <mat-form-field appearance="outline">
+                  <mat-select #designation formControlName="designation" i18n-placeholder
+                    placeholder="Select designation"
+                    (selectionChange)="otherDropDownChange(designation.value, 'designation')">
+                    <mat-option [value]="designation.name" *ngFor="let designation of designationsMeta">
+                      {{designation.name}}
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html]: please do optional check

File: project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html [hunk 3]
```diff
@@ -48,187 +41,240 @@
...
+                </mat-form-field>
+                <mat-icon class="verified-icon margin-left-xs">check_circle</mat-icon>
+              </div>
+
+              <div class="md:w-1/3">
+                <label class="margin-remove-bottom form-label required">Group</label>
+                <mat-form-field appearance="outline">
+                  <mat-select #group formControlName="group" i18n-placeholder placeholder="Select group">
+                    <mat-option [value]="g" *ngFor="let g of groupsList">
+                      {{g.name}}
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html:82]: same as above

File: project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html [hunk 4]
```diff
@@ -48,187 +41,240 @@
...
-              </mat-grid-tile>
+                </div> -->
+              </div>
+            </div>
+            <div class="flex flex-row margin-top-m">
+              <div class="md:w-1/3">
+                <label class="margin-remove-bottom form-label required">Date of Birth</label>
+                <mat-form-field appearance="outline" class="customdob">
+                  <input id="birthDate" matInput #dob formControlName="dob" [matDatepicker]="picker" i18n-placeholder
+                    (focus)="picker.open()" placeholder="Choose date of birth" [min]="1/1/1900" [max]="today">
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html:188]: please check for min date with product

File: project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html [hunk 5]
```diff
@@ -255,37 +300,79 @@
...
+                  <mat-list-option *ngFor="let role of rolesList" (click)="modifyUserRoles(role.roleName)"
+                    [selected]="userRoles.has(role.roleName)" [value]="role.roleName">
+                    {{role.roleName}}
+                  </mat-list-option>
+                </mat-selection-list>
+              </div>
+
+              <div class="md:w-1/2" *ngIf="isMdoAdmin">
+                <label class="view-text-label">Roles</label>
+                <div *ngIf="user.organisations && user.organisations[0].roles.length > 0">
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html]: please do optional check

File: project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html [hunk 6]
```diff
@@ -255,37 +300,79 @@
...
+                  <ul class="roleslist">
+                    <li *ngFor="let t of user.organisations[0].roles">{{t}}
+                    </li>
+                  </ul>
+                </div>
+                <p *ngIf="user.organisations && user.organisations[0].roles.length === 0">-</p>
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html]: same as above

File: project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html [hunk 7]
```diff
@@ -381,10 +470,192 @@
...
+</div>
+
+<!-- for Approvals page -->
+<div *ngIf="isApprovals">
+  <mat-accordion [multi]="true">
+    <mat-expansion-panel class="user-card" *ngFor="let appData of approvalData" #panel
+      (opened)="closeOtherPanels(panel)">
+      <mat-expansion-panel-header (click)="getUerData(appData.user)">
+        <div class="float-left" (click)="$event.stopPropagation();">
+          <div class="initialcircle display-inline-block margin-right-s" *ngIf="appData.user">
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html]: same as above

File: project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html [hunk 8]
```diff
@@ -381,10 +470,192 @@
...
+
+            <div class="md:w-1/2">
+              <label class="view-text-label">Tags</label>
+              <div *ngIf="appData.user.profileDetails && appData.user.profileDetails.additionalProperties &&
+                    appData.user.profileDetails.additionalProperties.tag &&
+                    appData.user.profileDetails.additionalProperties.tag.length > 0">
+                <ul class="tagslist">
+                  <li *ngFor="let t of appData.user.profileDetails.additionalProperties.tag">{{t}}</li>
+                </ul>
+              </div>
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html:640]: same as above

File: project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html [hunk 9]
```diff
@@ -381,10 +470,192 @@
...
+              <p *ngIf="appData.user.profileDetails && !appData.user.profileDetails.additionalProperties">-</p>
+            </div>
+          </div>
+          <div class="flex flex-row margin-top-m">
+            <div class="md:w-1/1">
+              <label class="view-text-label">Roles</label>
+              <div *ngIf="appData.user.organisations && appData.user.organisations[0].roles.length > 0">
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html]: same as above

File: project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html [hunk 10]
```diff
@@ -381,10 +470,192 @@
...
+                  <li *ngFor="let t of appData.user.organisations[0].roles">{{t}}
+                  </li>
+                </ul>
+              </div>
+              <p *ngIf="appData.user.organisations && appData.user.organisations[0].roles.length === 0">-</p>
+            </div>
+          </div>
         </div>
       </div>
+      <div class="panel-body" *ngIf="!appData.user">
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html]: please do optional check whereever needed

File: project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.ts [hunk 11]
```diff
@@ -75,16 +158,92 @@ export class UserCardComponent implements OnInit {
...
+  private filterLanguage(name: string) {
+    if (name) {
+      const filterValue = name.toLowerCase()
+      return this.masterLanguagesEntries.filter((option: any) => option.name.toLowerCase().includes(filterValue))
+    }
+    return this.masterLanguagesEntries
+  }
+
+  numericOnly(event: any): boolean {
+    const pattren = /^([0-9])$/
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.ts:232]: please use pattern as a variable

File: project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html (hunk 4 above)
Comment (NandiniAV) [reply project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html:188]: Have added same value as in  learner portal edit profile component

File: project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.ts (hunk 11 above)
Comment (NandiniAV) [reply project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.ts:232]: It s added as a event function check on paste & keypress

File: project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html [hunk 12]
```diff
@@ -381,14 +471,197 @@
...
+            <div class="md:w-1/4">
+              <label class="view-text-label">Category</label>
+              <p>{{appData.user.profileDetails && appData.user.profileDetails.personalDetails &&
+                appData.user.profileDetails.personalDetails.category ?
+                appData.user.profileDetails.personalDetails.category : '-'}}</p>
+            </div>
+
+            <div class="md:w-1/4">
+              <label class="view-text-label">Pay Band</label>
+              <p>{{appData.user.profileDetails && appData.user.profileDetails.employmentDetails &&
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/components/user-cards/user-card.component.html:614]: please do optional check

**Pull request merged**